src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -423,39 +398,36 @@ std::shared_ptr<SyncUser> SyncManager::get_current_user() const
     return cur_user_ident ? get_user_for_identity(*cur_user_ident) : nullptr;
 }
 
-void SyncManager::log_out_user(const std::string& user_id)
+void SyncManager::log_out_user(const SyncUser& user)
 {
     util::CheckedLockGuard lock(m_user_mutex);
 
     // Move this user to the end of the vector
-    if (m_users.size() > 1) {
```

Comment:
It's impossible to hit the case where this'd be a problem because SyncUser will serialize multithreaded logout/remove on a specific user and calling both at the same time won't result in this function being called if the other thread removed the user first, but I do think it makes sense to be defensive here anyway.